### PR TITLE
feat: Add auto-pass branch prefix feature

### PR DESCRIPTION
**Pull request by simpsonw**
## Description

This PR adds a new `auto-pass-branch-prefix` input parameter that allows the combined status check action to automatically pass (skip all status check validation) when the branch name starts with a configured prefix. This enables trusted automated branches to bypass status checks entirely.

The implementation includes:
- New `auto-pass-branch-prefix` input parameter in action.yml (opt-in with empty default)
- `getBranchFromContext()` function to extract branch names from GitHub events (pull requests, push events)
- Auto-pass logic in main() that checks branch prefix and exits early if matched
- Comprehensive unit tests for branch extraction (8 tests covering all event types)
- Integration tests for the complete auto-pass flow (3 end-to-end tests)
- User-facing documentation in README.md with security warnings

## Context / Why are we making this change?

Some automated systems (like AI coding agents or automated refactoring tools) create branches that should always be allowed to merge without waiting for full CI/CD pipeline completion. These branches are from trusted sources and don't require the same level of validation as human-authored changes.

By adding a branch prefix filter, teams can configure their workflows to automatically pass the combined status check for branches like `grimoire-*` or `automated/*`, while still requiring full status checks for all other branches. This provides a balance between automation velocity and code quality gates.

## Testing and QA Plan

The implementation includes comprehensive test coverage:

**Unit Tests (8 tests for `getBranchFromContext()`):**
- Extracts branch name from pull_request events
- Extracts branch name from push events with refs/heads/ format
- Handles branches with slashes in the name
- Returns null for tag events, release events, and undefined refs
- All tests use real GitHub event payload structures

**Integration Tests (3 end-to-end tests for main() flow):**
- ✅ Auto-passes when branch matches prefix (early exit, no status checks)
- ✅ Proceeds normally when branch doesn't match prefix
- ✅ Proceeds normally when prefix is not configured (feature disabled)

**All 26 tests passing** (existing tests + new tests)

Manual testing was performed by running the action in different scenarios:
- Branch `grimoire-test-123` with prefix `grimoire-` → immediate pass with appropriate log message
- Branch `feature-test` with prefix `grimoire-` → normal status check flow
- No prefix configured → no performance impact, normal flow

I also added a manual test verifying that it will pass if matching the auto-pass prefix (https://github.com/etsy/config-ramp-testing/actions/runs/21368248572/job/61505710513) and run the checks normally if it doesn't match the auto-pass prefix (https://github.com/etsy/config-ramp-testing/actions/runs/21368417260/job/61506297877).

## Impact

**Backward Compatibility:** This change is **100% backward compatible**. The `auto-pass-branch-prefix` parameter has an empty string default, meaning the feature is completely opt-in. Existing workflows will continue to work exactly as before with no changes required.

**Performance:** When the feature is disabled (default), there is zero performance impact—the branch check is skipped entirely if no prefix is configured.

**Security:** The README includes a prominent security warning advising users to only enable this feature for trusted automated systems. The case-sensitive prefix matching prevents accidental bypasses.

**Use Cases:**
- AI coding agents (e.g., Grimoire branches)
- Automated dependency updates from trusted bots
- Automated refactoring tools
- Internal automation systems with controlled branch creation

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/main.test.ts [hunk 1]
```diff
@@ -117,3 +121,346 @@ describe('RequiredCheckRunResult interface', () => {
...
+
+  afterEach(() => {
+    // Restore timers
+    jest.useRealTimers()
+
+    // Restore mocks
+    getInputMock.mockRestore()
+    infoMock.mockRestore()
+
+    // Restore context
```
[the remark this answers is not in the document]
Comment (simpsonw) [reply src/main.test.ts]: This should be addressed here: https://github.com/etsy/combined-status-check-action/pull/20/commits/4558b0cfdb1b19d3bc0c950d915c00c9e9b5d5fc

I also added a new test for when the auto-pass branch is null.

**Review by russtaylor — APPROVED** [review]
LGTM! ty!!!
